db/compaction_picker.cc

```diff
@@ -1373,6 +1356,39 @@ bool LevelCompactionBuilder::PickFileToCompact() {
   return start_level_inputs_.size() > 0;
 }
 
+bool LevelCompactionBuilder::PickL0ToBaseCompaction() {
```

Comment:
When you said, "then we only compact this L0 file with one L1 file and nothing more," did you really mean compact with zero L1 files since that's the requirement for trivial move? I think we can provide this optimization, too, let me try.